### PR TITLE
Canvas: Fix selection box glitch when adding certain elements

### LINKED ISSUE #91558: Vizards: Selecto Improvements
During selection of elements for ad-hoc viz creation, after creation of the ad-hoc viz, dragging the new viz element behaves strangely. I think we need to unselect after creation of ad-hoc viz.

https://github.com/user-attachments/assets/fef8868e-fd0c-4cfd-92f9-289745d1ce0e

### PR DESCRIPTION
**Pull request by nmarrs**
Fixes issue where selecto selection box would be drawn when moving certain newly added elements to the canvas. This issue was impacting: 
- `Ellipse`
- `Server`
- `Triangle`
- `Cloud`
- `Parallelogram`
- `Button`
- `Wind turbine`
- `All drone elements`

Before

https://github.com/user-attachments/assets/95f5fd71-8178-4b17-accb-9dcafd882475

After

https://github.com/user-attachments/assets/718f99ff-f298-40b3-bfaa-8bc10e8b408a

The way we were calculating whether or not a element was already selected was contingent on having a consistent DOM structure which is no longer the case.

Fixes https://github.com/grafana/grafana/issues/91558

### CODE REVIEW

File: public/app/features/canvas/runtime/sceneAbleManagement.ts [hunk 1]
```diff
@@ -33,6 +33,38 @@ const enableCustomables = (moveable: Moveable) => {
...
+  Given this, we need to traverse up the DOM tree from the selected target to find
+  the element's registered selecto div to determine if the selected target is already selected in selecto state.
+  See `initMoveable` and `generateTargetElements` for more context.
+*/
+const isTargetAlreadySelected = (selectedTarget: HTMLElement, scene: Scene) => {
+  let selectedTargetParent = selectedTarget.parentElement;
+  let isTargetAlreadySelected = false;
+
+  // Traverse up the DOM tree to check if the selected target is already selected
+  while (selectedTargetParent) {
```
Comment (drew08t) [inline public/app/features/canvas/runtime/sceneAbleManagement.ts:49]: This makes me nervous. I know it should break when we hit scene root, and beyond that at Document, but perhaps there is a more direct way to do this loop that involves a depth counter.

**Comment by drew08t** [comment]
Tested and behaves as expected. Other than my suggestion to consider, LGTM!

File: public/app/features/canvas/runtime/sceneAbleManagement.ts (hunk 1 above)
Comment (nmarrs) [reply public/app/features/canvas/runtime/sceneAbleManagement.ts:49]: where would we have this depth counter? 🤔 I'm not sure of a better way of handling this given inconsistent nesting / DOM structure of canvas elements

Comment (drew08t) [reply public/app/features/canvas/runtime/sceneAbleManagement.ts:49]: I thought maybe we already had one during grouping events.😳

I think your approach is okay, but can we test that it does in fact exit the while when reaching Document by removing the scene root break?

Comment (nmarrs) [reply public/app/features/canvas/runtime/sceneAbleManagement.ts:49]: in my testing it never reached edge case of going up to scene.root.div (mainly due to the current target is coming from a selecto drag start event which is are limited to canvas elements 🤔) These are initialized in the `initMovable` function via https://github.com/grafana/grafana/blob/1534329fe9a038c07cf3ead2ecdbe1049a882e84/public/app/features/canvas/runtime/sceneAbleManagement.ts#L71

**Review by drew08t — APPROVED** [review]

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. This is a friendly reminder that the Steady [11.2.x Release Branch](https://github.com/grafana/grafana/tree/v11.2.x) was cut this week, and that it includes the changes committed up to https://github.com/grafana/grafana/commit/b4126d3bce3627e9460e2db0e870c8d984be8e84. This PR was merged after it, so we wanted to let you know that you can add the `backport v11.2.x` label to to backport it and make it part of the next 11.2.x Patch Release, just in case you may have missed the news.

Thank you!

**Pull request merged**